pkg/cloudprovider/providers/aws/sets_ippermissions.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type IPPermissionSet map[string]*ec2.IpPermission
+
+func NewIPPermissionSet(items ...*ec2.IpPermission) IPPermissionSet {
```

Comment:
It wasn't - we can't currently autogenerate things that can't be map-keys.  We had the same thing in #21907 for sets of net.IPNet.  There we ended up putting it into pkg/util/net/sets, expected alias for the import being netsets, with a readme that we will likely move it to pkg/util/sets when/if we get autogen to work.  And the API for netsets.IPNet should match the API for sets, at least as much as is implemented.

This one is AWS only, so I don't think we should let it escape pkg/cloudproviders/providers/aws.